deltacat/aws/clients.py

```diff
@@ -183,10 +193,10 @@ def _client(name: str, region: Optional[str], **kwargs) -> BaseClient:
 def resource_cache(name: str, region: Optional[str], **kwargs) -> ServiceResource:
     # we don't use the @lru_cache decorator because Ray can't pickle it
     cached_function = lru_cache()(_resource)
-    return cached_function(name, region, **kwargs)
+    return boto_retry_wrapper(cached_function, name, region, **kwargs)
```

Comment:
I have a meta question. Why are we having to hit IMDS with high TPS when we are caching this resource?

Reply:
From the error stack traces, the throttling is occurring during cache misses.

Reply:
Ah.. Each worker is a separate process. May be that's why. I am wondering if we should instead be thinking of per-node cache. Thoughts?

Reply:
For per-node cache, are you thinking about a Ray actor? I've found that the boto resource or session can't be cached outside of it's process due to serialization issues.

Reply:
Yeah, we can use per-node memcached/redis/ray plasma. Feel free to pick one. We just have to make sure it's not transferred over the wire. Also, I could pickle session though.  

```
>>> s = boto3.Session()
>>> dmps = cloudpickle.dumps(s)
>>> loads = cloudpickle.loads(dmps)
```

Reply:
The boto resource isn't serializable, but yeah you're right that the session can be serialized with the Ray pickler at least (pymemcache's default serde didn't work for me). 

I played around with sessions using memcached, but I still find the throttling issue to persist for heavy workloads. From the error traces, it seems to be due to how there will always be a credentials check when we instantiate a new service client or resource, even from a re-used session.